crates/infra/utils/src/cargo/workspace.rs

```diff
@@ -99,12 +99,21 @@ impl CargoWorkspace {
     }
 
     pub fn get_command(subcommand: impl AsRef<str>) -> Result<Command> {
+        let subcommand = subcommand.as_ref();
+
         let mut command = Command::new("cargo")
-            .arg(subcommand.as_ref())
+            .arg(subcommand)
             .flag("--all")
-            .flag("--all-targets")
             .flag("--all-features");
 
+        if subcommand == "doc" {
```

Comment:
`get_command()` returns the command for further processing instead of executing it directly. Instead of special cases here, I suggest moving `--all-targets` to be appended by specific callers, and also moving `--no-deps` to be appended by `cargo doc` specifically.

Reply:
I knew that you'd not want to do that :laughing: I decided to do that in the first place to not specify much work explicitly per each subcommand but keep it here. In any case, I've changed this in 8aa5435c48838c345ea6d697fef100e4c0a425f3; let me know if that's enough!

Reply:
It is unfortunate that they are not consistent enough :( and that we have to override `rustflags` from the command line, and override it EXACTLY THE SAME on all Cargo invocations in order not to disturb incremental builds. But it is what it is 😞

Reply:
Thank you for doing this!